### PR TITLE
Initial stab at a GQL API and UI

### PR DESCRIPTION
**Pull request by orta**
This is a possible start based on the example inside [GraphQL Sequelize](https://github.com/mickhansen/graphql-sequelize/tree/master/examples)

I've ran out of time for today, but at least someone can continue from here rather than it be abandoned. running `yarn start` in the api package will launches the graphql, which has a working models section for both GQL and the db. It's set up to support associations fine with diamond dependencies in the current setup.

No pressure for this to be _the way_

### CODE REVIEW

File: packages/api/src/models/Group.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { sequelize } from '../db';
+import { User } from './User';
+
+class Group extends Model {
+  static tableName = 'groups';
+  static Creator = undefined;
+
+  static associate(models) {
+    console.log("-----")
+    // Group.Groups = Group.belongsTo(User, {
```
Comment (Guusy) [inline packages/api/src/models/Group.js]: :eye: can remove this comment ?

File: packages/api/src/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// @ts-check
```
Comment (orta) [inline packages/api/src/index.js]: Basically you get the tooling for TypeScript without using it

File: packages/api/src/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    users: {
+      type: userGroupsConnection.connectionType,
+      args: userGroupsConnection.connectionArgs,
+      resolve: userGroupsConnection.resolve
+    }
+  }
+})
+
+const schema = new GraphQLSchema({ query: root });
+const app = express();
```
Comment (orta) [inline packages/api/src/index.js]: Someone in chat said folks agreed against graphic, in which case this PR can be closed

File: packages/api/src/index.js (hunk 2 above)
Comment (shovon) [reply packages/api/src/index.js]: Nice. A great start.

**Comment by email2vimalraj** [comment]
I believe we should focus on putting the Open API specification first before diving into the code.

**Comment by vaibhavsingh97** [comment]
> I believe we should focus on putting the Open API specification first before diving into the code.

We can use swagger for that, should I open a PR with basic specification?

**Comment by QuincyLarson** [comment]
@orta Thanks for opening this. It's great to spike on this, though I agree we shouldn't merge any code until we've agreed the spec.

@email2vimalraj Would you agree with @vaibhavsingh97 that Swagger is the best place for this?

I'm closing this PR so it doesn't accidentally get merged, but we can continue the discussion here.

**Comment by Wizyma** [comment]
Do we really need swagger with GraphQl ? The documentation of GraphQl is autogenerated, and there is tool like graphql-voyager to visualise the data flow wich is way more friendly than swagger.

**Comment by email2vimalraj** [comment]
@QuincyLarson yes there is another PR which is opened by another member related to swagger. 
@vaibhavsingh97 it would be great if you can help us with that PR

**Pull request closed without merging**